### DISCUSSION TITLE
What would be the easiest way to implement a settings menu?

### DISCUSSION THREAD

**Discussion by revolter** [python-telegram-bot/python-telegram-bot#2596] [Q&A]
I don't have much experience with Telegram bots, so I would appreciate some pointers to implement the interface for a settings menu.

Should I use a `/settings` command, or a reply keyboard? Maybe `telegram.ext` already contains helpful utilities for this?

**Comment by Bibo-Joshi** [comment]
If you're talking only about toggle/select-options (in contrast to settings where the user has to make some input), IMO the cleanest user interface is achieved with inline keyboards. You can use a `ConversationHandler` for that, as showcased in `inlinekeyboard2.py`. Ofc, to enter the settings menu you need some sort of other input. Personally I'm not a fan of having a custom reply keyboard showing all the time and would rather go with a command, but that's probably a question of taste.
